app/src/main/java/org/mozilla/fenix/downloads/DownloadsFragment.kt

```diff
@@ -0,0 +1,129 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.downloads
+
+import android.content.Context
+import android.content.Context.MODE_PRIVATE
+import android.content.SharedPreferences
+import android.os.Bundle
+import android.util.Log
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.Fragment
+import androidx.recyclerview.widget.RecyclerView
+import com.google.gson.Gson
+import com.google.gson.reflect.TypeToken
+import kotlinx.android.extensions.LayoutContainer
+import kotlinx.android.synthetic.main.component_downloads.*
+import kotlinx.android.synthetic.main.download_list_item.*
+import mozilla.components.browser.state.state.content.DownloadState
+import org.mozilla.fenix.R
+import org.mozilla.fenix.ext.requireComponents
+
+class DownloadsFragment: Fragment() {
```

Comment:
If we are to bring a new Fragment it would be best to have it use the MVI architecture from the start - https://github.com/mozilla-mobile/fenix/blob/master/docs/architecture-overview.md

`BookmarkFragment`, `HistoryFragment` or `ShareFragment` can be used as an inspiration since they also have to display a list of items but they are not fully respecting the architecture.
We can do better.